src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/EditApprenticeshipCommandHandlerTests.cs

```diff
@@ -0,0 +1,332 @@
+﻿using AutoFixture;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.CommitmentsV2.Application.Commands.EditApprenticeship;
+using SFA.DAS.CommitmentsV2.Application.Queries.GetTrainingProgramme;
+using SFA.DAS.CommitmentsV2.Authentication;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using SFA.DAS.CommitmentsV2.Messages.Events;
+using SFA.DAS.CommitmentsV2.Models;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using SFA.DAS.CommitmentsV2.Types;
+using SFA.DAS.Testing.Builders;
+using SFA.DAS.UnitOfWork.Context;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.CommitmentsV2.UnitTests.Application.Commands
+{
+    public class EditApprenticeshipCommandHandlerTests
+    {
+        EditApprenticeshipCommandHandlerTestsFixture fixture;
+
+        [SetUp]
+        public void Setup()
+        {
+            fixture = new EditApprenticeshipCommandHandlerTestsFixture();
+        }
+
+        [Test]
+        public async Task OnlyImmediateUpdate_WhenOnlyEmployerReferenceIsChanged()
+        {
+            fixture.Command.EditApprenticeshipRequest.EmployerReference = "NewEmployerRef";
+
+            await fixture.Handle();
+            fixture.VerifyOnlyImmediateUpdate();
+        }
+
+        [Test]
+        public async Task WhenFirstNameIsChanged()
```

Comment:
Should these unit test names be "Then" rather than "When"?

Reply:
done